### PR TITLE
Build warning fix proposal, mostly about reference to packed fields.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: libafl/src/bolts/llmp.rs [hunk 1]
```diff
@@ -255,15 +255,15 @@ unsafe fn _llmp_page_init<SHM: ShMem>(shmem: &mut SHM, sender: u32, allow_reinit
     };
     (*page).magic = PAGE_INITIALIZED_MAGIC;
     (*page).sender = sender;
-    ptr::write_volatile(&mut (*page).current_msg_id, 0);
+    ptr::write_volatile(ptr::addr_of_mut!((*page).current_msg_id), 0);
```
Comment (domenukk) [inline libafl/src/bolts/llmp.rs:258]: That's a great catch! Everything should be nice and aligned, but this seems better. 
Where did the warnings come up btw?

File: libafl/src/bolts/serdeany.rs [hunk 2]
```diff
@@ -58,7 +58,7 @@ where
     where
         D: serde::de::Deserializer<'de>,
     {
-        let mut erased = erased_serde::Deserializer::erase(deserializer);
+        let mut erased = <dyn erased_serde::Deserializer>::erase(deserializer);
```
Comment (domenukk) [inline libafl/src/bolts/serdeany.rs:61]: @andreafioraldi you'll have to look at that one ;)

File: libafl/src/bolts/llmp.rs (hunk 1 above)
Comment (devnexen) [reply libafl/src/bolts/llmp.rs:258]: I use nightly mainly

Comment (domenukk) [reply libafl/src/bolts/llmp.rs:258]: So stable may get these warnings eventually, interesting

**Comment by domenukk** [comment]
Looks good to me, thanks! Waiting for @andreafioraldi before merging

**Pull request merged**